group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -869,6 +954,10 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
                     groupId, memberId, updatedMember.subscribedTopicRegex());
                 bumpGroupEpoch = true;
             }
+        } else {
+            if (staticMemberReplaced) {
+                records.add(newMemberSubscriptionRecord(groupId, updatedMember));
+            }
```

Comment:
If we rely on `member` and `updatedMember` then we don't need this because `!updatedMember.equals(member)` will catch the new member.

Reply:
Yes, that makes sense.